spec/solver_spec.rb

```diff
@@ -0,0 +1,27 @@
+# require_relative 'solver'
+require './solver'
+require 'rspec'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver
+  end
+
+  it 'returns factorial of a number' do
```

Comment:
[Optional] while you did an amazing job working on the requested changes I noticed you commented out some code in this file it isnt good to do this because it makes your code base look untidy it would be nice to remove this in all files were code has been commented ].